File: src/main/scala/info/crlog/higgs/util/Algorithms.scala
package info.crlog.higgs.util

import java.math.BigInteger
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

/**
 * Courtney Robinson <courtney@crlog.info>
 */
object Algorithms {
  def md5(str: String): String = {
    var m: MessageDigest = null
    try {
      m = MessageDigest.getInstance("MD5")
    }
    catch {
      case e: NoSuchAlgorithmException => {
      }
    }
    m.update(str.getBytes, 0, str.length)
    return new BigInteger(1, m.digest).toString(16)
  }

  def sha1(str: String): String = {
    var m: MessageDigest = null
    try {
      m = MessageDigest.getInstance("SHA-1")
    }
    catch {
      case e: NoSuchAlgorithmException => {
      }
    }
    m.update(str.getBytes, 0, str.length)
    return new BigInteger(1, m.digest).toString(16)
  }
}

File: higgs-core/src/main/java/com/fillta/higgs/util/Algorithms.java
package com.fillta.higgs.util;

import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

/**
 * @author Courtney Robinson <courtney@crlog.info>
 */
public class Algorithms {
    public static String sha256(String str) {
        try {
            MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
            messageDigest.update(str.getBytes());
            return new String(messageDigest.digest());
        } catch (NoSuchAlgorithmException e) {
            return null;
        }
    }
}
